torch/_dynamo/guards.py

```diff
@@ -511,6 +521,50 @@ def combine_scopes(left, right):
         self.check_fn = self.compile_check_fn(local_builder, global_builder)
         self._seen_ids.clear()
 
+    def _parse_symbolic_shape_expressions(self, tensor_check_names, tensor_check_ids):
```

Comment:
The right way to do this might be as a custom sympy printer.

Reply:
Blocking? Or for later?

Reply:
for later

Reply:
IDK, the current implementation looks buggy, I'd rather do it right with a printer (or, you don't have to use sympy, you can just traverse the AST recursively) to begin with

Reply:
Where does it look buggy?

Reply:
Updated with printer